consensus/safety-rules/src/tests/suite.rs

```diff
@@ -958,3 +964,86 @@ fn test_2chain_timeout(constructor: &Callback) {
         Error::NotSafeToTimeout(4, 1, 3, 2)
     );
 }
+
+/// Test that we can succesfully sign a valid commit vote
+fn test_sign_commit_vote(constructor: &Callback) {
+    // we construct a chain of proposals
+    // genesis -- a1 -- a2 -- a3
+
+    let (mut safety_rules, signer, key) = constructor();
+    let (proof, genesis_qc) = test_utils::make_genesis(&signer);
+
+    let round = genesis_qc.certified_block().round();
+    safety_rules.initialize(&proof).unwrap();
+
+    let a1 = test_utils::make_proposal_with_qc(round + 1, genesis_qc, &signer, key.as_ref());
+    let a2 = make_proposal_with_parent(round + 2, &a1, None, &signer, key.as_ref());
+    let a3 = make_proposal_with_parent(round + 3, &a2, Some(&a1), &signer, key.as_ref());
+
+    // now we try to agree on a1's execution result
+    let ledger_info_with_sigs = a3.block().quorum_cert().ledger_info();
+
+    assert!(safety_rules
+        .sign_commit_vote(
+            ledger_info_with_sigs.clone(),
+            ledger_info_with_sigs.ledger_info().clone()
+        )
+        .is_ok());
+
+    // check empty ledger info
+    assert!(safety_rules
+        .sign_commit_vote(
+            a2.block().quorum_cert().ledger_info().clone(),
+            a3.block().quorum_cert().ledger_info().ledger_info().clone()
+        )
+        .is_err());
+
+    // non-dummy blockinfo test
+    assert!(safety_rules
+        .sign_commit_vote(
+            LedgerInfoWithSignatures::new(
+                LedgerInfo::new(
+                    BlockInfo::new(
+                        ledger_info_with_sigs.ledger_info().commit_info().epoch(),
+                        ledger_info_with_sigs.ledger_info().commit_info().round(),
+                        ledger_info_with_sigs.ledger_info().commit_info().id(),
+                        ledger_info_with_sigs
+                            .ledger_info()
+                            .commit_info()
+                            .executed_state_id(),
+                        100,
+                        ledger_info_with_sigs
+                            .ledger_info()
+                            .commit_info()
+                            .timestamp_usecs(),
+                        None,
+                    ),
+                    ledger_info_with_sigs.ledger_info().consensus_data_hash()
+                ),
+                BTreeMap::<AccountAddress, Ed25519Signature>::new()
```

Comment:
this failed the signature check first?

Reply:
I change the order of the checks as checking signatures would be more expensive than value checkings.